expression/builtin_miscellaneous_vec.go

```diff
@@ -217,11 +218,36 @@ func (b *builtinIntAnyValueSig) vecEvalInt(input *chunk.Chunk, result *chunk.Col
 }
 
 func (b *builtinIsIPv4CompatSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinIsIPv4CompatSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	prefixCompat := []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			i64s[i] = 0
+		} else {
+			ipAddress := []byte(buf.GetString(i))
```

Comment:
We can use `buf.GetBytes(i)` to get `[]byte`

Reply:
Thanks for pointing out. It indeed speed up. I have updated the new benchmark result.